workflow/schemas.py

```diff
@@ -296,6 +335,12 @@ def _corners_to_array(corners_spec: list[dict[str, float]]) -> np.ndarray:
         Literal("resolution", description="Subdivision resolution."): And(
             float, _is_positive
         ),
+        Optional(
+            Literal(
+                "point_source_params",
+                description="Parameters for point source approximation, if applicable",
+            )
+        ): Or(None, POINT_SOURCE_PARAMS_SCHEMA),
```

Comment:
After putting point source params class in this module you can now write
```suggestion
        ): Or(None, And(POINT_SOURCE_PARAMS_SCHEMA, Use(PointSourceParams))),
}
```
and remove the `__post_init__` stuff.